### PR TITLE
Build in a basic HTTP client for Apple platforms

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

**Comment by swankjesse** [comment]
This does not work, not at all.

```
Test running process exited unexpectedly.
Current test: makeRequests
Process output:
 continuation: CancellableContinuation(Shareable[kotlin.native.concurrent.WorkerBoundReference@18b8368]){Active}@3b8608
b
Uncaught Kotlin exception: kotlin.native.IncorrectDereferenceException: illegal attempt to access non-shared <object>@1688b88 from other thread
    at 0   test.kexe                           0x104f9bf7f        kfun:kotlin.Throwable#<init>(kotlin.String?){} + 95 (/opt/buildAgent/work/67fbc2b507315583/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Throwable.kt:24:37)
    at 1   test.kexe                           0x104f95683        kfun:kotlin.Exception#<init>(kotlin.String?){} + 91 (/opt/buildAgent/work/67fbc2b507315583/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:23:44)
    at 2   test.kexe                           0x104f95837        kfun:kotlin.RuntimeException#<init>(kotlin.String?){} + 91 (/opt/buildAgent/work/67fbc2b507315583/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:34:44)
    at 3   test.kexe                           0x104fa7937        kfun:kotlin.native.IncorrectDereferenceException#<init>(kotlin.String){} + 91 (/opt/buildAgent/work/67fbc2b507315583/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/Runtime.kt:34:36)
    at 4   test.kexe                           0x104faa57b        ThrowIllegalObjectSharingException + 423 (/opt/buildAgent/work/67fbc2b507315583/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/concurrent/Internal.kt:115:11)
    at 5   test.kexe                           0x10510e523        _ZN12_GLOBAL__N_128throwIllegalSharingExceptionEP9ObjHeader + 27 
    at 6   test.kexe                           0x10510ed8f        _ZN12_GLOBAL__N_136terminateWithIllegalSharingExceptionEP9ObjHeader + 11 
    at 7   test.kexe                           0x105118977        _ZNK16KRefSharedHolder3refIL11ErrorPolicy3EEEP9ObjHeaderv + 111 
    at 8   test.kexe                           0x104e4ad23        _ZL39Kotlin_Interop_unwrapKotlinObjectHolderP11objc_object + 47 
    at 9   test.kexe                           0x104da4b1b        _6170702e636173682e7a69706c696e653a7a69706c696e652d6c6f61646572_knbridge3 + 183 (/Users/jwilson/Development/zipline/zipline-loader/src/nativeMain/kotlin/app/cash/zipline/loader/URLSessionZiplineHttpClient.kt:40:29)
    at 10  CFNetwork                           0x19b46c667        CFURLCredentialStorageCopyAllCredentials + 24387 
    at 11  CFNetwork                           0x19b3fe623        CFURLRequestCopyAllHTTPHeaderFields + 8879 
    at 12  libdispatch.dylib                   0x19658a5ef        _dispatch_call_block_and_release + 31 
    at 13  libdispatch.dylib                   0x19658c1b3        _dispatch_client_callout + 19 
    at 14  libdispatch.dylib                   0x1965938a7        _dispatch_lane_serial_drain + 667 
    at 15  libdispatch.dylib                   0x196594437        _dispatch_lane_invoke + 443 
    at 16  libdispatch.dylib                   0x19659ec97        _dispatch_workloop_worker_thread + 647 
    at 17  libsystem_pthread.dylib             0x19674c35f        _pthread_wqthread + 287 
    at 18  libsystem_pthread.dylib             0x19674b07f        start_wqthread + 7 
```

File: zipline-loader/src/nativeTest/kotlin/app/cash/zipline/loader/URLSessionZiplineHttpClientTest.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+package app.cash.zipline.loader
+
+import kotlin.test.Test
+import kotlinx.coroutines.runBlocking
+import platform.Foundation.NSURLSession
+
+class URLSessionZiplineHttpClientTest {
+  @Test
+  fun makeRequests(): Unit = runBlocking {
```
Comment (swankjesse) [inline zipline-loader/src/nativeTest/kotlin/app/cash/zipline/loader/URLSessionZiplineHttpClientTest.kt]: Just using this harness to see what happens. I need MockWebServer in multiplatform! 

(FYI @yschimke!)

File: zipline-loader/src/nativeTest/kotlin/app/cash/zipline/loader/URLSessionZiplineHttpClientTest.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertTrue
+import kotlinx.coroutines.runBlocking
+import okio.IOException
+import platform.Foundation.NSURLSession
+
+/**
+ * This tests our Kotlin/Native URLSessionZiplineHttpClient. Unfortunately this test is not enabled
+ * by default as we don't have an equivalent to MockWebServer for Kotlin/Native.
```
Comment (JakeWharton) [inline zipline-loader/src/nativeTest/kotlin/app/cash/zipline/loader/URLSessionZiplineHttpClientTest.kt:28]: ktor server + arraydeque of simple response sealed class will probably get us very far here

**Review by JakeWharton — APPROVED** [review]

File: zipline-loader/src/nativeTest/kotlin/app/cash/zipline/loader/URLSessionZiplineHttpClientTest.kt (hunk 2 above)
Comment (swankjesse) [reply zipline-loader/src/nativeTest/kotlin/app/cash/zipline/loader/URLSessionZiplineHttpClientTest.kt:28]: ooooh yeah that’s a great idea. 
https://github.com/cashapp/zipline/issues/715

**Pull request merged**